Relationship/config.txt

//================================================================================
// BASIC CONFIG
//================================================================================
	
// Current test name 
testName = map01

// Knowledge base local (1 = DBpedia; 2 = my knowledge base, in this case directedStreamGraph was changed to true)
knowledgeBasePlace = 2

// Base directory to creation of output files: absolut path 'C:\\path\\results' or relative path 'path\\results' or current path '.'
// baseDirectory = C:\\Users\\Henrique\\Documents\\data_collect
baseDirectory = C:\\documentos\\Doutorado\\Tese\\my_knowledge_base

// minimum quantity of iterations to get out of loop when connected component = 1 (good value: 6) 
minIterationToVerifyUniqueConnectedComponent = 6

// minimum quantity of iterations to get out of loop when exist relationship between original concepts (good value: 10)
minIterationToVerifyRelationshipBetweenOriginalConcepts = 8

// maximum quantity of iterations to get out of loop (good value: 50)  
maxIteration = 50  

// N-degree filter used in all system (good value: 2)
nDegreeFilter = 2

// apply n-degree filter trigger from iteration number (good value: 2)
iterationTriggerApplyNDegreeFilterAlgorithm = 3

// quantity of nodes to shoot n-degree filter algorithm (good value: 10000)
quantityNodesToApplyNdegreeFilter = 10000

// apply n-degree filter algorithm only if connected component = 1 (good value: true)
isUniqueConnectedComponentToApplyNdegreeFilter = false

// make the duplication of concept: from with "category" to without one (change to modify quantity of concepts with "category")
additionNewConceptWithoutCategory = true
	
// K-core used in all system (good value: 2)
kCoreFilter = 2

// quantity of nodes to shoot K-core n algorithm
quantityNodesToApplyKcoreFilter = 700
	
// fator added in logarithmic function to calculate goal of concepts quantity in final concept to map (good value: 3)
// function: log2(1/countOriginal)+4+factor
conceptsQuantityCalculationFactor = 0

// top limit to range of concepts quantity (good value: 5)
conceptsMinMaxRange = 5
	

//================================================================================
// ADVANCED CONFIG
//================================================================================

// Name of file with the concepts (it is recommended an amount of 2 to 12 concepts)
nameUserTermsFile = terms.txt

// use of the useless table to discart terms in rdfs collect fase
isEnableUselessTable = true
	
// BETWEENNESS + CLOSENESS
// quantity of nodes to selection in all connected componentes (proporcion relative to quantity total of user terms) (good value: 3.0)
proporcionBetweennessCloseness = 1.0
// precision added up to rounding the calculate of quantity of each connected component (good value: 0.5)
precisionBetweennessCloseness = 0.5
// quantity of nodes to use as base to build betweenness+closeness sorted table. 
// proporcion relative to quantity total of user terms. Must be >= maxBetweennessCloseness.  (good value: double of maxBetweennessCloseness)
// (if near maxBetweennessCloseness than the sort will disregard closeness) 
proporcionBetweennessOnly = 10
// maximum limit to quantity of new concepts distributed in all connected component, +0.5 in each component (excluding the addition of nodes by "Category:") 
// proporcion relative to quantity total of user terms. (good value: 5)
maxBetweennessCloseness = 5

// EIGENVECTOR
// quantity of nodes to selection in all connected componentes (proporcion relative to quantity total of original nodes) (good value: 1.0)
proporcionEigenvector = 1.0
// precision added up to rounding the calculate of quantity of each connected component (good value: 0.5)
precisionEigenvector = 0.5 
// maximum limit to quantity of new concepts distributed in all connected component, +0.5 in each component (excluding the addition of nodes by "Category:") 
// proporcion relative to quantity total of user terms. (good value: 15)
maxEigenvector = 5

// choice of nodes to be head. They are used to build the shortest paths (obs. the original concepts are always chosen) (good value: isBetweennessCloseness = true)
isBetweennessCloseness = true
isEigenvector = false
isSelected = false
	
// keep all nodes with link to original concepts (in stage after selection of head nodes) (good value: false)
// (normally this flag improves much more the final quantity of concepts in concept map) 
isKeepNeighborsOfOriginalConcepts = false
	
// quantity of nodes each rank to show in short report (good value: 0)
quantityNodesShortReport = 0

// color of original concepts (good value: 200.200.200.255)
backGroundcolorOriginalConcept = 200.200.200.255

// thickness of concepts with hint (good value: 2)
borderThicknessConceptWithHint = 2

// line maximum length of sentences in the final concept map
maxLineLengthConcept    = 15
maxLineLengthLinkPhrase = 15

// Names of files  (## = will be change to value of the testName var)
nameQueryDefaultFile    = query_model\\query.txt
nameVocabularyFile      = vocabulary\\linkvocabulary.txt
nameUselessConceptsFile = vocabulary\\uselessconcepts.txt
nameTxtConceptMapFile   = conceptmap_##.txt
nameCxlConceptMapFile   = conceptmap_##.cxl
nameGexfGraphFile       = graph_##.gexf
nameCompleteReportFile  = complete_report_##.txt
nameShortReportFile     = short_report_##.txt
nameConsoleReportFile   = console_report_##.txt
nameConsoleErrorFile    = consoleErr_##.txt
nameMyKnowledgeBaseFile = myKnowledgeBase\\base.txt

// Names of directories
dirRdfsPersistenceFiles = persistenceRdfs
dirGraph                = graph
dirLog					= log
dirConceptMap           = conceptMap    	
    	
// DBPEDIA server (good value: http://dbpedia.org/sparql) (alternative good value: http://lod.openlinksw.com/sparql)
dbpediaServer = http://dbpedia.org/sparql

// to see simultaneously the build of graph in Gephi (good value: false)	
gephiVisualization = false

// to see simultaneously the build of graph in Graph Stream Visualization (good value: false)	
graphStreamVisualization = 	false

// to fix a bug in Gephi Tool Kit - calculate wrong the quantity of connected component
// turn the flag to true and put original concepts that is getting alone (put '.' to empty)
isFixBugInGephiToolKit = false
originalConceptWithGephiToolKitBug1 = .
originalConceptWithGephiToolKitBug2 = .
originalConceptWithGephiToolKitBug3 = .

	


Relationship/src/main/Config.java
package main;
public interface Config {

	// ============================================================================
	// CONFIG - RUN, LOG AND PRINT 
    //
	String  testNumber = "00"; 
	
	String nameFileCompletReport = "C:\\Users\\Henrique\\Documents\\log\\complete_report_"+Config.testNumber+".txt";
	String nameFileShortReport   = "C:\\Users\\Henrique\\Documents\\log\\short_report_"+Config.testNumber+".txt";
	String nameFileConsoleReport = "C:\\Users\\Henrique\\Documents\\log\\console_report_"+Config.testNumber+".txt";
	String nameFileConsoleError  = "C:\\Users\\Henrique\\Documents\\log\\consoleErr.txt";
	String doubleLine = "\n=============================================================================================\n";
	String singleLine = "\n---------------------------------------------------------------------------------------------\n";
	String starsLine  = "\n*********************************************************************************************\n";

	// ============================================================================
	// CONFIG - INPUT FILES 
    //
	String nameUserTermsFile       = "userterms\\terms"+Config.testNumber+".txt";
	String nameQueryDefaultFile    = "query_model\\query.txt";
	String nameVocabularyFile      = "vocabulary\\linkvocabulary.txt";
	String nameUselessConceptsFile = "vocabulary\\uselessconcepts.txt";
	
	
	// ============================================================================
	// CONFIG - OUTPUT FILES  (.GEXF and .TXT concept map) 
    //
	String nameGexfGraphFile      = "graph\\graph"+Config.testNumber;
	String nameTxtConceptMapFile  = "C:\\Users\\Henrique\\Documents\\conceptmap"+Config.testNumber+".txt";
	String nameCxlConceptMapFile  = "C:\\Users\\Henrique\\Documents\\conceptmap"+Config.testNumber+".cxl";
    	
	// ============================================================================
	// CONFIG - RDFs 
    //
	String selectedServiceSnorqlEndPoint = Config.serviceSnorqlEndpoint_1;
	
	String serviceSnorqlEndpoint_1 = "http://dbpedia.org/sparql";
	String serviceSnorqlEndpoint_2 = "http://lod.openlinksw.com/sparql/";
	String serviceSnorqlEndpoint_3 = "http://www.ida.liu.se/projects/semtech/energy/snorql/";
	String serviceSnorqlEndpoint_4 = "http://data.nobelprize.org/snorql/";

	String addressBasic = "http://relationship/";
	String addressImage = "http://commons.wikimedia.org/wiki/File:";
	
	String markQueryReplacement = "#######";
	
	String originalConceptAddress = "http://dbpedia.org/resource/";
	String namePersistenceRDFsTableFile = "persistenceRdfs\\persistenceRdfs.dat";
	String dirRdfsPersistenceFiles = "persistenceRdfs";
	
	// ============================================================================
	// CONFIG - GRAPHs 
    //
	String nameGraph = "Graph relationship";

	boolean gephiVisualization = 		false;
	String nameGephiWorkspace = 		"workspace0";

	boolean graphStreamVisualization = 	false;

	int maxNodes = 50000;
	int minEdges = 50000;

	boolean ignoreCaseConcept = false;
	
	boolean edgeLabelStreamGephi = true;
	boolean nodeLabelStreamGephi = false; // original concepts always have label in Stream Gephi
	
	boolean edgeLabelFileGephi   = true;
	boolean nodeLabelFileGephi   = false; // original concepts always have label in Stream Gephi

	boolean directedStreamGraph = false;
	boolean directedGephiGraph  = true;  // only accept true 
 
	
	// ============================================================================
	// CONFIG - ANALYSIS 
    //
	// range of the quantity of iterations
	int minIterationToVerifyUniqueConnectedComponent             = 6;   // better set: 7
	int minIterationToVerifyRelationshipBetweenOriginalConcepts  = 10;   // better set: 10
	// it is necessary a great value to ensure that the connected component = 1 or level relationship between original concepts = 0  
	int maxIteration = 50;  
	
	// proporcion above total original concept (used in the build betweenness+closeness sorted table) 
	double proporcionBetweenness = 10;

	// quantity of nodes to selection in all connected componentes (proporcion relative to quantity total of original nodes)
	double proporcionBetweennessCloseness = 2.0;
	// precision added up to rounding the calculate of quantity of each connected component
	double precisionBetweennessCloseness = 0.5;
	// maximum limit to quantity of new concepts distributed in all connected component, +0.5 in each component (excluding the addition of nodes by "Category:")
	double maxBetweennessCloseness = 15;

	// quantity of nodes to selection in all connected componentes (proporcion relative to quantity total of original nodes)
	double proporcionEigenvector = 2.0;
	// precision added up to rounding the calculate of quantity of each connected component
	double precisionEigenvector = 0.5; 
	// maximum limit to quantity of new concepts distributed in all connected component, +0.5 in each component (excluding the addition of nodes by "Category:")
	double maxEigenvector = 15; 

	// make the duplication of concept: from with "Category:" to without one
	boolean additionNewConceptWithoutCategory = true;
	
	// indicate concepts that do not belong to connected component (for example: original conpects)
	int withoutConnectedComponent = -1;
	
	// K-core used in all system
	int kCoreFilter = 2;
	// since iteration number x to apply K-core filter trigger
	int iterationTriggerApplyKCoreFilterAlgorithm = 5;  // better set: 5
	// quantity of nodes to shoot K-core n algorithm
	int quantityNodesToApplyKcoreFilter = 700;
	
	// N-degree filter used in all system
	int nDegreeFilter = 2;
	// since iteration number x to apply n-degree filter trigger
	int iterationTriggerApplyNDegreeFilterAlgorithm = 1;  // better set: 2
	// quantity of nodes to shoot n-degree filter algorithm
	int quantityNodesToApplyNdegreeFilter = 5000;  // better set: 20000
	
	// choice of nodes to be head (they are used to build the shortest paths)
	// obs.: the original concepts are always chosen
	boolean isBetweennessCloseness = true;
	boolean isEigenvector = false;
	boolean isSelected = false;
	
	// keep all nodes with link to original concepts (in stage after selection of head nodes)
	// normally this flag improves much more the final quantity of concepts in concept map 
	boolean isKeepNeighborsOfOriginalConcepts = false;
	
	// to fix a bug in Gephi Tool Kit - calculate wrong the quantity of connected component
	// turn the flag to true and put the especific original concept that is getting alone
	boolean isFixBugInGephiToolKit = true;
	String originalConceptWithGephiToolKitBug = "Tim Berners-Lee";
	
	// ============================================================================
	// CONFIG - REPORT 
    //
	// quantity of nodes to show in short report
	int quantityNodesShortReport = 1;

	// ============================================================================
	// CONFIG - CONCEPT MAP 
    //
	// to calculate min and max concepts to map
	int conceptsQuantityCalulationFactor = 18;  // (factor default: 18) - goal of concepts quantity: log2(1/original quantity)*2 + (factor)
	int conceptsMinMaxRange = 5;
	
	String backGroundcolorOriginalConcept = "200,200,200,255";
	String borderThicknessConceptWithHint = "2";
	
}
